frontend/src/App.tsx

```diff
@@ -1,21 +1,53 @@
-import {BrowserRouter as Router, Routes, Route, Navigate} from "react-router-dom";
-import SignupPage from './components/SignupPage';
-import {Box} from "@mui/material";
+import {
+  BrowserRouter as Router,
+  Routes,
+  Route,
+  Navigate,
+} from "react-router-dom";
+import SignupPage from "./components/SignupPage";
+import { Box } from "@mui/material";
 import React from "react";
+import { UserProvider } from "./context/UserContext";
+import LoginPage from "./components/LoginPage";
+import HomePage from "./components/HomePage";
+import Navbar from "./components/Navbar";
+import { useUser } from "../src/context/UserContext";
 
 function App() {
-    return (
-        <div className="App">
-            <Box display={"flex"} flexDirection={"column"} padding={"4rem"}>
-                <Router>
-                    <Routes>
-                        <Route path="/" element={<Navigate replace to="/signup" />}></Route>
-                        <Route path="/signup" element={<SignupPage/>}/>
-                    </Routes>
-                </Router>
-            </Box>
-        </div>
-    );
+  
+  const user = useUser();
+
+  return (
+    <div className="App">
+        <Box display={"flex"} flexDirection={"column"} padding={"4rem"}>

```

Comment:
I think its better to add the stylings into the sx prop that material UI provides: https://mui.com/material-ui/react-box/